src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -154,19 +157,11 @@ export type Fiber = {
   // component. This indicates whether it is better to continue from the
   // progressed work or if it is better to continue from the current state.
   progressedPriority: PriorityLevel,
-
-  // If work bails out on a Fiber that already had some work started at a lower
-  // priority, then we need to store the progressed work somewhere. This holds
-  // the started child set until we need to get back to working on it. It may
-  // or may not be the same as the "current" child.
-  progressedChild: Fiber | null,
-
-  // When we reconcile children onto progressedChild it is possible that we have
-  // to delete some child fibers. We need to keep track of this side-effects so
-  // that if we continue later on, we have to include those effects. Deletions
-  // are added in the reverse order from sibling pointers.
-  progressedFirstDeletion: Fiber | null,
-  progressedLastDeletion: Fiber | null,
+  progressedWork: ProgressedWork,
+  // A pooled ProgressedWork object, allocated once per fiber pair. Should not
+  // be accessed outside of this module.
+  _pooledProgressedWork: null | ProgressedWork,
+  newestWork: Fiber,
```

Comment:
I have a comment in beginWorl but yeah I'll add one here, too. I'm planning to do another comments pass before landing.